VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseDiscussionResponsesAdapter.java

```diff
@@ -292,8 +294,24 @@ public void onException(Exception ex) {
         holder.socialLayoutViewHolder.threadFollowContainer.setVisibility(View.INVISIBLE);
 
         if (comment.isEndorsed()) {
+            final IAuthorData endorser = new IAuthorData() {
```

Comment:
This can be moved to a getter method in `DiscussionComment` (e.g. `getEndorsedByData()` or `getEndorser()`), which would return null if it's not endorsed.